include/openssl/thread.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 1995-2022 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2002, Oracle and/or its affiliates. All rights reserved
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OPENSSL_THREAD_H
+# define OPENSSL_THREAD_H
+
+# define OSSL_THREAD_SUPPORT_FLAG_THREAD_POOL (1U<<0)
+# define OSSL_THREAD_SUPPORT_FLAG_DEFAULT_SPAWN (1U<<1)
+
+# include <stdint.h>
+
+uint32_t OSSL_get_thread_support_flags(void);
```

Comment:
Wouldn't `uint64_t` be a better choice (gives us more flags)

Reply:
The thread pool design document says `uint32_t`, I wanted to follow it to the letter.

Reply:
OK...but we might want to reconsider at some point before we release - but lets not hold up this PR on that discussion.